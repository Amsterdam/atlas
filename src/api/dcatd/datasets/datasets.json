{
  "dct:title": "Afval inzamelingsdagen huisvuil en grofvuil",
  "dct:description": "Afval ophaalgebieden/inzamelingsdagen als gebieden voorzien van:\n- van/tot tijdstip van aanbieden\n- aanbiedwijze\n- frequentie van ophalen in dagen\n- stadsdeel\n\nDeze shapefiles wordt ook gebruikt voor de geosearch API voor afvalophaalgebieden waar je X,Y co\u00f6rdinaten aan kan geven om middels het het ophaalgebied de bijbehorende aanbiedwijze terug te krijgen binnen de Gemeente Amsterdam.\n \nDit zal de Inzamelingsdagen API blijven tot de nieuwe Omgevingswet dezelfde informatie landelijk zal verstrekken.",
  "ams:status": "beschikbaar",
  "dcat:distribution": [
    {
      "dct:title": "Inzamelingsdagen huisvuil",
      "dct:description": "Gezipte shapefile opgeslagen als RD new (EPSG:28992) co\u00f6rdinatenstelsel",
      "dcat:accessURL": "https://e85bcf2124fb4437b1bc6eb75dfc3abf.objectstore.eu/dcatd/bb511c25d44b4f058a39dbea46cc4020",
      "ams:purl": "https://api.data.amsterdam.nl/dcatd/datasets/qji2W_HBpWUWyg/purls/1",
      "ams:resourceType": "data",
      "ams:distributionType": "file",
      "dcat:mediaType": "application/x-zipped-shp",
      "dct:modified": "2020-11-02",
      "dc:identifier": "1",
      "ams:classification": "public",
      "dcat:byteSize": 571484,
      "foaf:isPrimaryTopicOf": { "dct:issued": "2019-07-22", "dct:modified": "2019-11-05" },
      "dct:license": "cc-zero",
      "@id": "_:d1"
    },
    {
      "dct:title": "Inzamelingsdagen grofvuil",
      "dct:description": "Gezipte shapefile opgeslagen als RD new (EPSG:28992) co\u00f6rdinatenstelsel",
      "dcat:accessURL": "https://e85bcf2124fb4437b1bc6eb75dfc3abf.objectstore.eu/dcatd/43873c7996a24f4f8e97490b37cfdf58",
      "ams:purl": "https://api.data.amsterdam.nl/dcatd/datasets/qji2W_HBpWUWyg/purls/2",
      "ams:resourceType": "data",
      "ams:distributionType": "file",
      "dcat:mediaType": "application/x-zipped-shp",
      "dct:modified": "2020-11-02",
      "dc:identifier": "2",
      "ams:classification": "public",
      "dcat:byteSize": 348402,
      "foaf:isPrimaryTopicOf": { "dct:issued": "2019-07-22", "dct:modified": "2019-11-05" },
      "dct:license": "cc-zero",
      "@id": "_:d2"
    },
    {
      "dct:title": "Afvalophaalgebieden API",
      "dcat:accessURL": "https://api.data.amsterdam.nl/afvalophaalgebieden",
      "ams:purl": "https://api.data.amsterdam.nl/dcatd/datasets/qji2W_HBpWUWyg/purls/3",
      "ams:resourceType": "app",
      "ams:distributionType": "api",
      "ams:serviceType": "rest",
      "dct:modified": "2019-07-22",
      "dc:identifier": "3",
      "ams:classification": "public",
      "dcat:byteSize": 0,
      "foaf:isPrimaryTopicOf": { "dct:issued": "2019-07-22", "dct:modified": "2019-07-22" },
      "dct:license": "cc-zero",
      "@id": "_:d3"
    }
  ],
  "overheidds:doel": "Het voorzien van actuele gegevens voor de afvalwijzer op basis van adres een co\u00f6rdinaat op een gebied te kunnen prikken om zo de afvaldagen te kunnen teruggeven.",
  "foaf:isPrimaryTopicOf": { "dct:issued": "2019-07-22", "dct:modified": "2019-07-22" },
  "dct:accrualPeriodicity": "irreg",
  "dct:temporal": {},
  "ams:temporalUnit": "na",
  "ams:spatialDescription": "Gemeente Amsterdam",
  "ams:spatialUnit": "specific",
  "dct:language": "lang1:nl",
  "ams:owner": "Gemeente Amsterdam, Afval en Grondstoffen",
  "dcat:contactPoint": { "vcard:fn": "Pieter Born", "vcard:hasEmail": "p.born@amsterdam.nl" },
  "dct:publisher": {
    "foaf:name": "Datapunt Amsterdam",
    "foaf:mbox": "datapunt@amsterdam.nl",
    "foaf:homepage": "https://data.amsterdam.nl"
  },
  "dcat:theme": ["theme:duurzaamheid-en-milieu"],
  "dcat:keyword": ["Afval", "Afvalophaaldagen", "Garbage collection"],
  "ams:license": "cc-zero",
  "ams:sort_modified": "2020-11-02",
  "ams:modifiedby": "r.van.barneveld@amsterdam.nl",
  "@context": {
    "ams": "http://datacatalogus.amsterdam.nl/term/",
    "ckan": "https://ckan.org/terms/",
    "class": "ams:class#",
    "dc": "http://purl.org/dc/elements/1.1/",
    "dcat": "http://www.w3.org/ns/dcat#",
    "dct": "http://purl.org/dc/terms/",
    "foaf": "http://xmlns.com/foaf/0.1/",
    "lang1": "http://id.loc.gov/vocabulary/iso639-1/",
    "lang2": "http://id.loc.gov/vocabulary/iso639-2/",
    "org": "ams:org#",
    "overheid": "http://standaarden.overheid.nl/owms/terms/",
    "overheidds": "http://standaarden.overheid.nl/owms/terms/ds#",
    "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
    "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
    "void": "http://rdfs.org/ns/void#",
    "skos": "http://www.w3.org/2004/02/skos/core#",
    "theme": "ams:theme#",
    "time": "http://www.w3.org/2006/time#",
    "vcard": "http://www.w3.org/2006/vcard/ns#",
    "ams:purl": { "@type": "@id" },
    "dcat:dataset": { "@container": "@list" },
    "dcat:distribution": { "@container": "@set" },
    "dcat:keyword": { "@container": "@set" },
    "dcat:landingpage": { "@type": "@id" },
    "dcat:theme": { "@container": "@set", "@type": "@id" },
    "dct:issued": { "@type": "xsd:date" },
    "dct:language": { "@type": "@id" },
    "dct:modified": { "@type": "xsd:date" },
    "foaf:homepage": { "@type": "@id" },
    "foaf:mbox": { "@type": "@id" },
    "vcard:hasEmail": { "@type": "@id" },
    "vcard:hasURL": { "@type": "@id" },
    "vcard:hasLogo": { "@type": "@id" },
    "ams-dcatd": "https://api.data.amsterdam.nl/dcatd/datasets/"
  },
  "overheid:authority": "overheid:Amsterdam",
  "@id": "ams-dcatd:qji2W_HBpWUWyg",
  "dct:identifier": "qji2W_HBpWUWyg"
}
