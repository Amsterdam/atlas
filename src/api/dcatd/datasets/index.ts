import fixtureSingle from './dataset.json'
import { DcatDataset } from './types'

export default fixtureSingle as DcatDataset
export * from './getDatasetById'
export * from './types'
