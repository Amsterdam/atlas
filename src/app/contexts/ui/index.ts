export { default, UiProvider, useIsEmbedded } from './UiContext'
