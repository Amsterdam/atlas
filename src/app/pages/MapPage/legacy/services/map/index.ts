export * from './fetchDetails'
export * from './getDetailUrl'
export * from './getServiceDefinition'
