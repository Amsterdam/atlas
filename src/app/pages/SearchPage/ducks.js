import paramsRegistry from '../../../store/params-registry'

export const REDUCER_KEY = 'SEARCH'
const SEARCH_SET_FILTERS = `${REDUCER_KEY}/SEARCH_SET_FILTERS`
const SEARCH_SET_QUERY = `${REDUCER_KEY}/SET_QUERY`

const initialState = {
  activeFilters: {},
  query: '',
}

function reducer(state = initialState, action) {
  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action),
  }

  switch (action.type) {
    case SEARCH_SET_FILTERS:
      return {
        ...enrichedState,
        activeFilters: action.payload,
      }

    case SEARCH_SET_QUERY:
      return {
        ...enrichedState,
        query: action.payload,
      }

    default:
      return enrichedState
  }
}

export const setQuery = payload => ({
  type: SEARCH_SET_QUERY,
  payload,
})

export const setActiveFilters = payload => ({
  type: SEARCH_SET_FILTERS,
  payload,
})

export const getQuery = ({ [REDUCER_KEY]: { query } }) => query
export const getActiveFilters = ({ [REDUCER_KEY]: { activeFilters } }) => activeFilters

export { reducer }
