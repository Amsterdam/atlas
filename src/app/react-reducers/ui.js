/* eslint-disable no-case-declarations */
export const REDUCER_KEY = 'ui'
export const SET_BACKDROP = `${REDUCER_KEY}/SET_BACKDROP`

export const initialState = {
  backdropKeys: [],
}

export default (state = initialState, action) => {
  switch (action.type) {
    // Increase or decrease the number of times the backdrop is triggered.
    // We do this so other triggers of show / hide backdrops won't overrule others while the
    // backdrop is active
    case SET_BACKDROP:
      const { key, open } = action.payload
      return {
        ...state,
        backdropKeys: state.backdropKeys.includes(key)
          ? state.backdropKeys.filter(backdropTrigger =>
              !open ? backdropTrigger !== key : backdropTrigger,
            )
          : [...state.backdropKeys, ...(open ? [key] : [])],
      }
    default:
      return state
  }
}

export const actions = {
  setBackDrop: SET_BACKDROP,
}
