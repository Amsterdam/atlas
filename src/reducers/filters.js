export function emptyFilters() {
  return {
    type: {
      id: 'EMPTY_FILTERS'
    }
  };
}
