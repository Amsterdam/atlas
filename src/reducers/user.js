export const USER_AUTHENTICATED = 'USER_AUTHENTICATED';
export const USER_SCOPES = 'USER_SCOPES';
export const USER_NAME = 'USER_NAME';

const initialState = {
  authenticated: false,
  scopes: {},
  name: ''
};

export default function UserReducer(state = initialState, action) {
  switch (action.type) {
    case 'USER_AUTHENTICATED':
      return {
        ...state,
        user: {
          ...state.user,
          authenticated: action.authenticated
        }
      };

    case 'USER_SCOPES':
      return {
        ...state,
        user: {
          ...state.user,
          scopes: action.scopes.reduce((acc, scope) => {
            acc[scope] = true;
            return acc;
          }, {})
        }
      };

    case 'USER_NAME':
      return {
        ...state,
        user: {
          ...state.user,
          name: action.name
        }
      };

    default:
      return state;
  }
};

window.UserReducer = UserReducer;
