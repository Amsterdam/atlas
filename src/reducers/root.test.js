import rootReducer from './root';

import * as AutoSuggestReducer from '../header/ducks/auto-suggest/auto-suggest';
import * as DataSelectionReducer from '../shared/ducks/data-selection/reducer';
import * as Datasets from '../shared/ducks/datasets/datasets';
import * as DataSelectionCatalogReducer
  from '../shared/ducks/datasets/apiSpecification/apiSpecification';
import * as ErrorMessageReducer from '../shared/ducks/error/error-message';
import * as UiReducer from '../shared/ducks/ui/ui';
import * as UserReducer from '../shared/ducks/user/user';
import * as MapDetailReducer from '../map/ducks/detail/map-detail';
import * as MapReducer from '../map/ducks/map/reducer';
import * as MapLayersReducer from '../map/ducks/layers/map-layers';
import * as MapBaseLayersReducer from '../map/ducks/base-layers/map-base-layers';
import * as MapPanelLayersReducer from '../map/ducks/panel-layers/map-panel-layers';
import * as PanoramaReducer from '../panorama/ducks/reducer';
import * as PanoPreviewReducer from '../panorama/ducks/preview/panorama-preview';
import * as FiltersReducer from '../shared/ducks/filters/filters';
import * as DataSearchReducer from '../shared/ducks/data-search/reducer';
import * as SelectionReducer from '../shared/ducks/selection/selection';
import * as DetailReducer from '../shared/ducks/detail/reducer';
import * as Content from '../shared/ducks/content/reducer';

describe('the root reducer', () => {
  AutoSuggestReducer.default = () => 'autoSuggest';
  DataSelectionReducer.default = () => 'dataSelection';
  DataSelectionCatalogReducer.default = () => 'catalogFilters';
  ErrorMessageReducer.default = () => 'error';
  UiReducer.default = () => 'ui';
  UserReducer.default = () => 'user';
  DetailReducer.default = () => 'detail';
  MapDetailReducer.default = () => 'mapDetail';
  MapReducer.default = () => 'map';
  MapLayersReducer.default = () => 'layers';
  MapBaseLayersReducer.default = () => 'baseLayers';
  MapPanelLayersReducer.default = () => 'panelLayers';
  PanoramaReducer.default = () => 'panorama';
  PanoPreviewReducer.default = () => 'panoramaPreview';
  DataSearchReducer.default = () => 'dataSearch';
  SelectionReducer.default = () => 'selection';
  FiltersReducer.default = () => 'filter';
  Datasets.default = () => 'datasets';
  Content.default = () => 'content';

  it('combines many reducers', () => {
    const state = {};
    const action = {};

    const output = rootReducer(() => 'location')(state, action);
    expect(output).toMatchSnapshot();
  });
});
