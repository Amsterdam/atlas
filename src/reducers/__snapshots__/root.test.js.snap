// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the root reducer combines many reducers 1`] = `
Object {
  "dataSearch": "dataSearch",
  "dataSelection": "dataSelection",
  "datasets": "datasets",
  "detail": "detail",
  "error": "error",
  "files": Object {
    "fileName": "",
    "fileUrl": "",
    "type": "default",
  },
  "filter": "filter",
  "location": "location",
  "map": "map",
  "mapDetail": "mapDetail",
  "mapLayers": Object {
    "baseLayers": "baseLayers",
    "layers": "layers",
    "panelLayers": "panelLayers",
  },
  "panorama": "panorama",
  "panoramaPreview": "panoramaPreview",
  "search": Object {
    "activeFilters": Array [],
    "page": 1,
    "sort": null,
  },
  "selection": "selection",
  "ui": "ui",
  "user": "user",
}
`;
